pkg/minikube/cni/cni.go

```diff
@@ -183,3 +197,33 @@ func applyManifest(cc config.ClusterConfig, r Runner, f assets.CopyableFile) err
 
 	return nil
 }
+
+// configureCNI - to avoid conflicting CNI configs, it sets:
+// - for crio: 'cni_default_network' config param via cni.Network
+// - for containerd and docker: kubelet's '--cni-conf-dir' flag to custom CNI Config Directory path (same used also by CNI Deployment).
+// ref: https://github.com/kubernetes/minikube/issues/10984 and https://github.com/kubernetes/minikube/pull/11106
+// Note: currently, this change affects only Kindnet CNI (and all multinodes using it), but it can be easily expanded to other/all CNIs if needed.
+// Note2: Cilium does not need workaround as they automatically restart pods after CNI is successfully deployed.
+func configureCNI(cc *config.ClusterConfig, cnm Manager) error {
+	if _, kindnet := cnm.(KindNet); kindnet {
```

Comment:
on the crio KVM test I see these failing https://storage.googleapis.com/minikube-builds/logs/11189/cc9afff/KVM_Linux_crio.html

```


214 | TestNetworkPlugins/group/auto/HairPin | 0.24
-- | -- | --
237 | TestNetworkPlugins/group/calico/DNS | 395.35
241 | TestNetworkPlugins/group/false/KubeletFlags | 0.24
252 | TestNetworkPlugins/group/flannel/DNS | 377.81


```

@ilya-zuyev do u midn taking a look ?